### ISSUE TITLE
Apply incremental amounts of blur to the picture, with hotkeys

### ISSUE THREAD

**Issue by JorgeR81** [mpv-player/mpv#8167]
Hi,

What I would like to do is to apply (“add”) incremental amounts of (Gaussian) blur to the video picture, with hotkeys, that I could assign in the “intput.conf” file.

You don’t need to create the hotkeys. I can assign them myself later. Like in this other feature request (“options: add --video-scale-x/y”).
https://github.com/mpv-player/mpv/issues/6303
https://github.com/mpv-player/mpv/commit/5f49009849271ecd995f9f41e4eb2a36a0fc024a

Reason for this:

I have lots of old SD quality video files, from DVD and VHS. 
They still look pretty good on my 27 inch CRT TV, but on a modern display you can see all the encoding “imperfections”.
I’ve found that if I apply a really small amount of blur, I get a pretty decent image quality, even in a modern display. The encoding artifacts are gone, and I don’t really lose a lot of detail. 

I can do this in Potpayer. But I've just found MPV, and I’m really happy with it. I’m ready to switch players. This is the last thing I wasn’t able to achieve yet.

Thanks
Jorge Rodrigues

**Comment by CounterPillow** [comment]
You can either use the libavfi ["gblur"](https://ffmpeg.org/ffmpeg-filters.html#gblur) filter with `--vf` for CPU filtering, or you can write a user shader which applies a gaussian blur on the GPU. There are many examples of gaussian blur shaders online, you'll just have to adapt them for mpv's hook system.

**Comment by JorgeR81** [comment]
Thanks for your quick answer!

Can you give me a few more directions about using the "gblur" filter? I can’t find it on MPV manual.
https://mpv.io/manual/master/#video-filters

Is it possible to hook it up to the hotkeys (or scroll wheel)?

Is that any chance this could be implemented directly in MPV, in a similar way to the commit I mentioned on my feature request, like zooming or panning the picture.

I’m completely new to MPV. I've, literally, download it on Saturday, set it put on Sunday, and joined Github on Monday...

**Comment by JorgeR81** [comment]
See, what I had in mind is to have available something like "video-blur", so that I could put the respective shortcut command in the input.conf file. Something like:

Shift+WHEEL_UP		    add video-blur   0.1
Shift+WHEEL_DOWN          add video-blur  -0.1

But I don't know if this makes any sense... I've only written VBA code for my Excel databases, nothing related to this stuff.

**Comment by Akemi** [comment]
that's something a script should do and nothing mpv will add.
see the docs for scripting:
https://github.com/mpv-player/mpv/blob/master/DOCS/man/lua.rst
https://github.com/mpv-player/mpv/blob/master/DOCS/man/javascript.rst

gblur is is an ffmpeg filter, see the docs CP linked.

**Comment by JorgeR81** [comment]
OK, thanks.

**Comment by JorgeR81** [comment]
Hi,

I'm learning as I go along ... so these might be some really dumb questions.  

Can I use this code to apply the gblur filter to a video played with MPV?https://hhsprings.bitbucket.io/docs/programming/examples/ffmpeg/blurring_unsharping/gblur_gaussian_blur_.html

Could this be done just with a lua script (on the "portable_config" folder) like this one?
(I'm just referring to this one because it's the first that I tested, to see how they work).
https://github.com/jgreco/mpv-scripts/blob/master/fastforward.lua

Does MPV come with the necessary ffmpeg libraries to apply the filter?
Or do I have to download ffmpeg?
https://ffmpeg.org/download.html

(I'm on Windows 10)

Thanks

**Comment by Akemi** [comment]
mpv is linked against ffmpeg, yes it will have the necessary requirements to use the gblur filter.
yes you can use a script to set a filter, **mp.set_property_native("vf","gblur=sigma=0.5:steps=1")**
yes you can apply that filter with mpv, **--vf=gblur=sigma=0.5:steps=1**.

**Comment by JorgeR81** [comment]
Thank you very much,
( I wasn't sure you were able to see new comments, after the issue is closed)

I was planning to do something very simple for my first script, to get the basics, and then focus my attention on the blur script, after better reading the documentation.
I was thinking about a script to automatically mute volume, when playback is not at normal speed.
I didn't find an option for that in the user manual (but didn't look very closely).

I was going to use some parts of the fast forward script as an example for that.

So my question right now would be:
Is there already an option to do this, or a script from another user to achieve the same result?
I don't want to waste time creating something that is already available.

**Comment by garoto** [comment]
Perhaps consider joining the [IRC channel](https://mpv.io/community/), where you're free to elucubrate about the many player behaviors you don't understand yet. Better leave the issue tracker for direct, straightforward questions or actual bug reporting.

**Comment by JorgeR81** [comment]
Yes, I was having the feeling that this was not the best place to continue this discussion.

I actually clicked on these links before opening these issues.
But Chrome doesn't open anything. Do I have to change my browser settings or something?

**Comment by garoto** [comment]
Just open https://webchat.freenode.net/ and follow the prompts.

**Comment by JorgeR81** [comment]
OK, thanks.

**Comment by JorgeR81** [comment]
Hi, 
I'm already on the Users Channel, but I was hoping I could still post here, about the blur filter script, because I made some progress and I have more specific questions about this.

I gave it a label and I'm using vf toggle, to turn the filter on an off.

The lua scrips is:
mp.set_property_native("vf","@sdblur:!gblur=sigma=0.5:steps=1")
--vf=@sdblur:!gblur=sigma=0.5:steps=1

In the mvp.config file I have:
vf-add=@sdblur:!gblur

In the input.conf file I have:
b vf toggle @sdblur

But I would like to change the sigma during playback, with the scroll wheel.
Can I use something like this?

Shift+WHEEL_UP vf-command @sdblur add sigma 0.1
Shift+WHEEL_DOWN vf-command @sdblur add sigma -0.1

And if this is possible, would sigma=0 (or <0) effectively "toggle off" the filter?
Or do I have to use a command in the script in to turn off the filter. 
vf remove @ sdblur:!gblur

Thanks
